stubs/oauthlib/oauthlib/common.pyi

```diff
@@ -1,64 +1,81 @@
-from _typeshed import Incomplete
-from typing import Any
+import re
+from collections.abc import Iterable, Mapping, Sequence
+from logging import Logger
+from typing import Any, AnyStr, Final, Literal, TypeVar, overload
+from typing_extensions import TypeAlias
 
-UNICODE_ASCII_CHARACTER_SET: str
-CLIENT_ID_CHARACTER_SET: str
-SANITIZE_PATTERN: Any
-INVALID_HEX_PATTERN: Any
-always_safe: str
-log: Any
+_T = TypeVar("_T")
+_K = TypeVar("_K")
+_V = TypeVar("_V")
 
-def quote(s, safe: bytes = b"/"): ...
-def unquote(s): ...
-def urlencode(params): ...
-def encode_params_utf8(params): ...
-def decode_params_utf8(params): ...
+_HTTPMethod: TypeAlias = Literal["CONNECT", "DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT", "TRACE"]
 
-urlencoded: Any
+UNICODE_ASCII_CHARACTER_SET: Final[str]
+CLIENT_ID_CHARACTER_SET: Final[str]
+SANITIZE_PATTERN: Final[re.Pattern[str]]
+INVALID_HEX_PATTERN: Final[re.Pattern[str]]
+always_safe: Final[str]
+log: Logger
 
-def urldecode(query): ...
-def extract_params(raw): ...
-def generate_nonce(): ...
-def generate_timestamp(): ...
-def generate_token(length: int = 30, chars=...): ...
+def quote(s: AnyStr, safe: bytes = b"/") -> str: ...
+def unquote(s: AnyStr) -> str: ...
```

Comment:
`AnyStr` is a type var, so using it only once in a function signature makes no sense. This should probably use `str | bytes` instead. (Same may be true of other signatures below, I didn't check yet.)

Reply:
got it, I'll fix it. I thought I could shorten the `str | bytes` entry this way

Reply:
This is a common misconception, unfortunately. And I actually think it works, but it's not really correct.